ORT scanner execution stops if file upload fails

In case ORT scan is instructed to upload file(s) execution stops if an exception is thrown. 

Expected behavior: Log exception and continue execution

**Example**
```
[DefaultDispatcher-worker-3] INFO  org.ossreviewtoolkit.scanner.provenance.DefaultPackageProvenanceResolver - Resolved revision for package 'NuGet:Microsoft.Win32.SystemEvents:Microsoft.Win32.SystemEvents:7.0.0' to d099f075e45d2aa6007a22b71b45a08758559f80 based on guessed revision d099f075e45d2aa6007a22b71b45a08758559f80.
[main] INFO  org.ossreviewtoolkit.scanner.Scanner - Resolved provenance for 1 package(s) in 35.547589626s.
[main] INFO  org.ossreviewtoolkit.scanner.Scanner - Resolving nested provenances for 1 package(s).
[DefaultDispatcher-worker-3] INFO  org.ossreviewtoolkit.scanner.provenance.DefaultNestedProvenanceResolver - Could not find a stored nested provenance for RepositoryProvenance(vcsInfo=VcsInfo(type=Git, url=https://github.com/dotnet/runtime.git, revision=d099f075e45d2aa6007a22b71b45a08758559f80, path=), resolvedRevision=d099f075e45d2aa6007a22b71b45a08758559f80), attempting resolution.
[DefaultDispatcher-worker-3] INFO  org.ossreviewtoolkit.downloader.vcs.Git - Updating working tree from 'https://github.com/dotnet/runtime.git'.
[DefaultDispatcher-worker-3] INFO  org.ossreviewtoolkit.downloader.vcs.Git - Trying to fetch only revision 'd099f075e45d2aa6007a22b71b45a08758559f80' with depth limited to 50.
[DefaultDispatcher-worker-3] INFO  org.ossreviewtoolkit.utils.common.ProcessCapture - Running 'git checkout d099f075e45d2aa6007a22b71b45a08758559f80' in '/tmp/ort-DefaultWorkingTreeCache10975042919976521339'...
[main] INFO  org.ossreviewtoolkit.scanner.Scanner - Resolved nested provenance for 1 package(s) in 2.423594956s.
[main] INFO  org.ossreviewtoolkit.scanner.Scanner - Reading stored scan results for 1 package(s) with 1 provenance(s).
[main] INFO  com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Starting...
[main] INFO  com.zaxxer.hikari.pool.HikariPool - HikariPool-1 - Added connection org.postgresql.jdbc.PgConnection@677349fb
[main] INFO  com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Start completed.
[main] INFO  org.ossreviewtoolkit.scanner.Scanner - Read the following stored scan result(s) in 6.235722541s:
[main] INFO  org.ossreviewtoolkit.scanner.Scanner - 	ScanCode: Result(s) for 1 of 1 provenance(s).
[main] INFO  org.ossreviewtoolkit.scanner.Scanner - Scanning RepositoryProvenance(vcsInfo=VcsInfo(type=Git, url=https://github.com/dotnet/runtime.git, revision=d099f075e45d2aa6007a22b71b45a08758559f80, path=), resolvedRevision=d099f075e45d2aa6007a22b71b45a08758559f80) (1 of 1)...
[main] INFO  org.ossreviewtoolkit.scanner.Scanner - Skipping RepositoryProvenance(vcsInfo=VcsInfo(type=Git, url=https://github.com/dotnet/runtime.git, revision=d099f075e45d2aa6007a22b71b45a08758559f80, path=), resolvedRevision=d099f075e45d2aa6007a22b71b45a08758559f80) (1 of 1) as all scanners have results.
[main] INFO  org.ossreviewtoolkit.scanner.Scanner - Creating file lists for 1 provenances.
[DefaultDispatcher-worker-3] INFO  org.ossreviewtoolkit.scanner.Scanner - Creating file list for provenance 0 of 1.
[DefaultDispatcher-worker-3] ERROR org.ossreviewtoolkit.model.utils.FileProvenanceFileStorage - Could not read from 6ec955334d75214b7db34e276d705ce16c2b5272/file_list: FileNotFoundException: /home/ort/.ort/scanner/file-lists/6ec955334d75214b7db34e276d705ce16c2b5272/file_list (No such file or directory)
Caused by: FileNotFoundException: /home/ort/.ort/scanner/file-lists/6ec955334d75214b7db34e276d705ce16c2b5272/file_list.xz (No such file or directory)
[DefaultDispatcher-worker-3] INFO  org.ossreviewtoolkit.downloader.vcs.Git - Updating working tree from 'https://github.com/dotnet/runtime.git'.
[DefaultDispatcher-worker-3] INFO  org.ossreviewtoolkit.downloader.vcs.Git - Trying to fetch only revision 'd099f075e45d2aa6007a22b71b45a08758559f80' with depth limited to 50.
[DefaultDispatcher-worker-3] INFO  org.ossreviewtoolkit.utils.common.ProcessCapture - Running 'git checkout d099f075e45d2aa6007a22b71b45a08758559f80' in '/tmp/ort-DefaultWorkingTreeCache10975042919976521339'...
[main] INFO  org.ossreviewtoolkit.scanner.Scanner - Created file list for 1 provenances in 55.235841579s.
[main] INFO  org.ossreviewtoolkit.scanner.Scanner - Creating file archives for 1 package(s).
[main] INFO  org.ossreviewtoolkit.downloader.vcs.Git - Updating working tree from 'https://github.com/dotnet/runtime.git'.
[main] INFO  org.ossreviewtoolkit.downloader.vcs.Git - Trying to fetch only revision 'd099f075e45d2aa6007a22b71b45a08758559f80' with depth limited to 50.
[main] INFO  org.ossreviewtoolkit.utils.common.ProcessCapture - Running 'git checkout d099f075e45d2aa6007a22b71b45a08758559f80' in '/tmp/ort-DefaultWorkingTreeCache10975042919976521339'...
[main] INFO  org.ossreviewtoolkit.model.utils.FileArchiver - Archiving files matching [**] from '/tmp/ort-DefaultProvenanceDownloader83847565494157134'...
[main] INFO  org.ossreviewtoolkit.model.utils.FileArchiver - Archived directory '/tmp/ort-DefaultProvenanceDownloader83847565494157134' in 40.073667143s.
Exception in thread "main" java.io.IOException: Could not store file at 'https://**********.execute-api.eu-west-1.amazonaws.com/prod/files/6ec955334d75214b7db34e276d705ce16c2b5272/archive.zip': 413 - Request Entity Too Large
	at org.ossreviewtoolkit.utils.ort.storage.HttpFileStorage.write(HttpFileStorage.kt:132)
	at org.ossreviewtoolkit.model.utils.FileProvenanceFileStorage.putData(FileProvenanceFileStorage.kt:63)
	at org.ossreviewtoolkit.model.utils.FileArchiver.archive(FileArchiver.kt:93)
	at org.ossreviewtoolkit.scanner.Scanner.createMissingArchives(Scanner.kt:657)
	at org.ossreviewtoolkit.scanner.Scanner.scan(Scanner.kt:179)
	at org.ossreviewtoolkit.scanner.Scanner$scan$4.invokeSuspend(Scanner.kt)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:280)
	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:85)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59)
	at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38)
	at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
	at org.ossreviewtoolkit.plugins.commands.scanner.ScannerCommand.runScanners(ScannerCommand.kt:227)
	at org.ossreviewtoolkit.plugins.commands.scanner.ScannerCommand.run(ScannerCommand.kt:154)
	at com.github.ajalt.clikt.parsers.Parser.parse(Parser.kt:198)
	at com.github.ajalt.clikt.parsers.Parser.parse(Parser.kt:211)
	at com.github.ajalt.clikt.parsers.Parser.parse(Parser.kt:18)
	at com.github.ajalt.clikt.core.CliktCommand.parse(CliktCommand.kt:400)
	at com.github.ajalt.clikt.core.CliktCommand.parse$default(CliktCommand.kt:397)
	at com.github.ajalt.clikt.core.CliktCommand.main(CliktCommand.kt:415)
	at com.github.ajalt.clikt.core.CliktCommand.main(CliktCommand.kt:440)
	at org.ossreviewtoolkit.cli.OrtMainKt.main(OrtMain.kt:67)
```

**Configuration**
```
  scanner:
    archive:
      enabled: true
    
      fileStorage:
        httpFileStorage:
          url: ${API_URL}
          headers:
            Content-Type: application/zip # ORT does not set a content-type but API GW requires it
```

--- a/scanner/src/main/kotlin/Scanner.kt
+++ b/scanner/src/main/kotlin/Scanner.kt
@@ -20,6 +20,7 @@
 package org.ossreviewtoolkit.scanner
 
 import java.io.File
+import java.io.IOException
 import java.nio.file.StandardCopyOption
 import java.time.Instant
 
@@ -714,20 +715,23 @@ class Scanner(
 
         val duration = measureTime {
             provenancesWithMissingArchives.forEach { (pkg, nestedProvenance) ->
-                runCatching {
-                    downloadRecursively(nestedProvenance)
-                }.onSuccess { dir ->
+                var dir: File? = null
+
+                // runCatching has a bug with smart-cast, see https://youtrack.jetbrains.com/issue/KT-62938.
+                try {
+                    dir = downloadRecursively(nestedProvenance)
                     archiver.archive(dir, nestedProvenance.root)
-                    dir.safeDeleteRecursively(force = true)
-                }.onFailure {
+                } catch (e: IOException) {
                     controller.addIssue(
                         pkg.id,
                         Issue(
-                            source = "Downloader",
+                            source = "Scanner",
                             message = "Could not create file archive for " +
-                                "'${pkg.id.toCoordinates()}': ${it.collectMessages()}"
+                                "'${pkg.id.toCoordinates()}': ${e.collectMessages()}"
                         )
                     )
+                } finally {
+                    dir?.safeDeleteRecursively(force = true)
                 }
             }
         }

